newdle/client/src/client.js

```diff
@@ -31,13 +31,13 @@ class Client {
     return this._request(flask`api.users`({q}));
   }
 
-  createNewdle(title, duration, timezone, timeslots, participants) {
+  createNewdle = (title, duration, timezone, timeslots, participants) => {
```

Comment:
That's why I avoided passing `client.something` in the hook's API I suggested on MM - I don't think having to rely on bound functions is really nice

Reply:
It's true that it's not super pretty, but in any case it's only here and it's for a good reason.

Reply:
I mean, another option could be to pass the client object as well, since it's the structure which contains the token. TBH I don't like much the way `client.js` works ☹️ It's not explicit what you are getting once you import a function, and that is (IMO) confusing.

Reply:
Actually I think when using separate requests you'll need to pass a promise directly to the hook's function instead of just the args so you can do something like this:

```
_answerNewdle(async () => {
  const {partCode} = await client.createParticipant();
  return await client.saveParticipantAnswers(partCode, ...);
});
```

That way the hook can switch to submitting state, then await on the promise from the async function, and set the result returned by it in the end.

I think errors also also properly propagate to the try..catch inside the hook.

Reply:
We could as well store the tokens in redux.

Reply:
I'd try to avoid putting data from the URL into redux ;) that gets very quickly very messy

Reply:
> I mean, another option could be to pass the client object as well,

The disadvantage there is that now you don't have a direct reference to the function anymore, so editors don't know that the (string?) name of the function passed to the hook is that particular function...

Reply:
Wait... the token is already in Redux. Why do we even need this Client object to be stateful?